packages/astro/src/core/build/index.ts

```diff
@@ -72,10 +73,13 @@ class AstroBuilder {
     );
     this.viteConfig = viteConfig;
     const viteServer = await vite.createServer(viteConfig);
+    // Listen to the server so that we can make fetch requests if needed.
+    await viteServer.listen(this.port);
```

Comment:
If the port is taken, `viteServer` should claim the next open one. It might be safer to update `this.port`, e.g.:

```js
  await viteServer.listen(this.port);
  this.port = viteServer.port;
```